src/analysis/Environments/DeleteButton.ts

```diff
@@ -1,36 +1,46 @@
 import _ from 'lodash/fp';
 import { ReactNode } from 'react';
 import { h } from 'react-hyperscript-helpers';
-import { isResourceDeletable } from 'src/analysis/utils/resource-utils';
 import { getDisplayRuntimeStatus } from 'src/analysis/utils/runtime-utils';
 import { Link } from 'src/components/common';
 import { makeMenuIcon } from 'src/components/PopupTrigger';
 import { isApp } from 'src/libs/ajax/leonardo/models/app-models';
 import { LeoRuntimeStatus } from 'src/libs/ajax/leonardo/models/runtime-models';
+import * as Utils from 'src/libs/utils';
 
-import { DecoratedComputeResource } from './Environments.models';
+import { DecoratedComputeResource, LeoResourcePermissionsProvider } from './Environments.models';
+
+type DeletePermissionsProvider = Pick<LeoResourcePermissionsProvider, 'canDeleteApp' | 'canDeleteResource'>;
 
 export interface DeleteButtonProps {
   resource: DecoratedComputeResource;
+  permissions: DeletePermissionsProvider;
   onClick: (resource: DecoratedComputeResource) => void;
 }
 
 export const DeleteButton = (props: DeleteButtonProps): ReactNode => {
-  const { resource, onClick } = props;
+  const { resource, permissions, onClick } = props;
   const resourceType = isApp(resource) ? 'app' : 'runtime';
-  const isDeletable = isResourceDeletable(resourceType, resource);
+  const isDeletable =
+    resourceType === 'app' ? permissions.canDeleteApp(resource) : permissions.canDeleteResource(resourceType, resource);
```

Comment:
We should use `isApp` here instead of a string. 

See the TODO in `isResourceDeletable` in the `ts` file, we shouldn't need to pass `resourceType` to `canDeleteResource`, we can ask the type of the `resource` directly.